src/parser.rs

```diff
@@ -64,7 +64,7 @@ named!(
             >> char!('-')
             >> day: flat_map!(take!(2), parse_to!(i32))
             >> space
-            >> hour: flat_map!(take!(2), parse_to!(i32))
+            >> hour: flat_map!(take_until!(":"), parse_to!(i32))
             >> char!(':')
             >> minute: flat_map!(take!(2), parse_to!(i32))
```

Comment:
This looks a little bit odd. Why should some device format the hour without a leading zero and the minute not? Same for the seconds.

Reply:
Fixed.